Update readme (#587)
microsoft/json-schemas: README.md

# Microsoft JSON schema publishing repository

This repository contains schemas which are published at https://developer.microsoft.com/json-schemas/. Maintenance and publishing is performed by Microsoft employees from different organizations. 

> [!IMPORTANT]
> You cannot access the schemas directly from https://developer.microsoft.com/json-schemas/. They are simply hosted there based on the folder structure in this repository. Tools are pointing automatically to the exact schema files as needed.

Last updated: 27 May 2026

## Publish Microsoft Json schemas to the centralized json schema repo

> Microsoft launched a centralized GitHub repo for publishing Microsoft-owned json schemas. Previously, json schema files were published to a public site (http://json-schema.org) for universally hosting json-schema files and to other Microsoft sites such as dev.office.com. However, there's an inherent risk in using the public site and much benefit to customers to provide a single, supported location for all Microsoft json schemas.

Reviewer: JSON - capitalize here and other places

```suggestion
Microsoft launched a centralized GitHub repo for publishing Microsoft-owned JSON schemas. Previously, JSON schema files were published to a public site (http://json-schema.org) for universally hosting json-schema files and to other Microsoft sites such as dev.office.com. However, there's an inherent risk in using the public site and much benefit to customers to provide a single, supported location for all Microsoft JSON schemas.
```

- Repo: https://github.com/Microsoft/json-schemas
- Admins: https://github.com/orgs/Microsoft/teams/json-schemas-admins
- Public site: https://developer.microsoft.com/json-schemas
- Owner: Vesa Juvonen vesaj@microsoft.com (Eastern European Time (EET))

> - Publishing: Self-publish via an admin from your team/org. 

Reviewer: ```suggestion
- Publishing: Self-publish via an admin from your team/organization.
```

- For publishing support or if you don't have an admin who can publish:
  - M365 Developer program engineering team: DevAppPortalEng@microsoft.com
  - Office Developer documentation team: opcontent@microsoft.com

>   - Check the json-schemas-admins repo for someone in your org with Admin perms!

Reviewer: ```suggestion
  - Check the json-schemas-admins repo for someone in your organization with Administrator permissions!
```

    
## Get admin perms REQUIRED

All Microsoft teams are welcome to publish schemas to this repo. Pick two team members and request admin perms. Submit your request to @Microsoft/json-schemas-admins for approval and cc: Vesa Juvonen. This may take some time, so please be sure to do this part in advance. 

After you're approved as an admin, you have the power to create your own folder, add your schemas, manage your pull requests, etc. We suggest you contact your documentation team or several reliable PMs/EMs on your team and request Admin perms for them as well. There is no central person (was lindalu) to publish and assist you. 

## Publish a new schema

> 1. Create one folder per team in the repo. Keep your folder structure as flat as reasonable. Each folder will become an element of the json URL (e.g. https://developer.microsoft.com/json-schemas/folder/schema_name.json) Use examples in the repo for guidance. Stay consistent.

Reviewer: ```suggestion
1. Create one folder per team in the repo. Keep your folder structure as flat as reasonable. Each folder will become an element of the JSON URL (e.g. https://developer.microsoft.com/json-schemas/folder/schema_name.json) Use examples in the repo for guidance. Stay consistent.
```

- lockfile-explorer https://graphprodblobstorage.blob.core.windows.net/content/json-schemas/en-us/html/lockfile-explorer/
- office-js https://graphprodblobstorage.blob.core.windows.net/content/json-schemas/en-us/html/office-js/
- pnp https://graphprodblobstorage.blob.core.windows.net/content/json-schemas/en-us/html/pnp/
- rig-package https://graphprodblobstorage.blob.core.windows.net/content/json-schemas/en-us/html/rig-package/
- rush https://graphprodblobstorage.blob.core.windows.net/content/json-schemas/en-us/html/rush/
- sp https://graphprodblobstorage.blob.core.windows.net/content/json-schemas/en-us/html/sp/
- spfx-build https://graphprodblobstorage.blob.core.windows.net/content/json-schemas/en-us/html/spfx-build/
- spfx https://graphprodblobstorage.blob.core.windows.net/content/json-schemas/en-us/html/spfx/
- teams-toolkit https://graphprodblobstorage.blob.core.windows.net/content/json-schemas/en-us/html/teams-toolkit/
- teams https://graphprodblobstorage.blob.core.windows.net/content/json-schemas/en-us/html/teams/
- tsdoc https://graphprodblobstorage.blob.core.windows.net/content/json-schemas/en-us/html/tsdoc/

> [!IMPORTANT]

> > If you add a new root level folder then you need to send an email to devappportaleng@microsoft.com (developer app portal engineering team) and request they "whitelist" your folder for removing locale from the full URL. You don't need to do this for sub-folders. As mentioned, this is the team that hosts the Azure blob so reach out to them for support if needed. You'll find they're very responsive if you catch them on India time! 

Reviewer: ```suggestion
> If you add a new root level folder then you need to send an email to devappportaleng@microsoft.com (developer app portal engineering team) and request they "allowlist" your folder for removing locale from the full URL. You don't need to do this for sub-folders. As mentioned, this is the team that hosts the Azure blob so reach out to them for support if needed. You'll find they're very responsive if you catch them on India time! 
```
